src/EmotiBitWiFiHost.cpp

```diff
@@ -59,73 +98,84 @@ void EmotiBitWiFiHost::getAvailableSubnets() {
+	if (availableNetworks.size() != currentavailableNetworks.size()) { //print all Networks whenever new Networks are detected
+		string allAvailableNetworks;
+		for (int network = 0; network < availableNetworks.size(); network++) {
+			allAvailableNetworks += "[" + availableNetworks.at(network) + ".*] ";
 		}
-		ofLogNotice() << "All Subnet(s): " << allAvailableSubnets;
+		ofLogNotice() << "All Network(s): " << allAvailableNetworks;
 	}
 }
 
-void EmotiBitWiFiHost::pingAvailableSubnets() {
-	getAvailableSubnets(); // Check if new subnet appeared after oscilloscope was open (e.g. a mobile hotspot)
+void EmotiBitWiFiHost::pingAvailableNetworks() {
+	getAvailableNetworks(); // Check if new network appeared after oscilloscope was open (e.g. a mobile hotspot)
 	string packet = EmotiBitPacket::createPacket(EmotiBitPacket::TypeTag::HELLO_EMOTIBIT, advertisingPacketCounter++, "", 0); 
 	string ip;
-	if (enableBroadcast) { // ToDo: add enableBroadcast case if it's determined to be desirable
-		advertisingCxn.SetEnableBroadcast(true);
-	}
-	if (emotibitSubnets.size() == 0) { //initial search through all subnets
-		for (int sub = 0; sub < availableSubnets.size(); sub++) {
-			if (enableBroadcast) { 
-				ip = availableSubnets.at(sub) + "." + ofToString(255);
+
+	if (emotibitNetworks.size() == 0) { //initial search through all Networks
+		for (int network = 0; network < availableNetworks.size(); network++) {
+			if (_hostAdvSettings.enableBroadcast) {
+				advertisingCxn.SetEnableBroadcast(true);
+				ip = availableNetworks.at(network) + "." + ofToString(255);
 				advertisingCxn.Connect(ip.c_str(), advertisingPort);
 				advertisingCxn.Send(packet.c_str(), packet.length());
 			}
 			else {
-				for (int hostId = 1; hostId < 255; hostId++) {
-					ip = availableSubnets.at(sub) + "." + ofToString(hostId);
+				advertisingCxn.SetEnableBroadcast(false);
+				for (int hostId = _hostAdvSettings.unicastIpRange.first; hostId < _hostAdvSettings.unicastIpRange.second; hostId++) {
+					ip = availableNetworks.at(network) + "." + ofToString(hostId);
 					advertisingCxn.Connect(ip.c_str(), advertisingPort);
 					advertisingCxn.Send(packet.c_str(), packet.length());
 				}
 			}
 		}
 	} 
-	else { // Once an EmotiBit is found, advertising is directed at that subnet to avoid network spam
-		for (int hostId = 1; hostId < 255; hostId++) {
-			ip = emotibitSubnets.at(0) + "." + ofToString(hostId);
+	else { // Once an EmotiBit is found, advertising is directed at that network to avoid network spam
+		if (_hostAdvSettings.enableBroadcast) {
+			advertisingCxn.SetEnableBroadcast(true);
+			ip = emotibitNetworks.at(0) + "." + ofToString(255);
 			advertisingCxn.Connect(ip.c_str(), advertisingPort);
 			advertisingCxn.Send(packet.c_str(), packet.length());
 		}
+		else {
+			advertisingCxn.SetEnableBroadcast(false);
+			for (int hostId = _hostAdvSettings.unicastIpRange.first; hostId < _hostAdvSettings.unicastIpRange.second; hostId++) {
```

Comment:
should be 

hostId <`=` _hostAdvSettings.unicastIpRange.second 

according to the `ipMax` value in emotibitCommSettings JSON file.
I think changing the code makes more sense since ipMax implies that the address is inclusive of that value.